Loop/Managers/DeviceDataManager.swift

```diff
@@ -1210,6 +1211,36 @@ extension DeviceDataManager {
                 case .cancelTemporaryOverride:
                     log.default("Canceling temporary override from remote command")
                     loopManager.settings.scheduleOverride = nil
+                case .bolusEntry(let bolusValue):
+                    log.default("Enacting remote bolus entry: %{public}@", String(describing: bolusValue))
+                    
+                    //Remote bolus requires validation from its remote source
+                    if remoteDataServicesManager.validatePushNotificationSource(notification) == false {
+                        log.info("Could not validate notification: %{public}@", String(describing: notification))
+                        return
+                    }
+
+                    // enact bolus; make sure maxbolus is in place for protection
+                    if let maxBolus = loopManager.settings.maximumBolus {
+                       if bolusValue.isLessThanOrEqualTo(maxBolus) {
+                          self.enactBolus(units: bolusValue, automatic: true)
+                       } else {
+                           log.default("Remote bolus higher than maximum. Aborting...")
+                       }
+                    } else {
+                        log.default("No max bolus detected. Aborting...")
+                    }
+                case .carbsEntry(let newEntry):
+                    log.default("Adding carbs entry.")
```

Comment:
A local notification should happen with this too.

Reply:
Updated